src/Domain/Creators/BinaryCompactObjectHelpers.cpp

```diff
@@ -41,30 +41,37 @@ create_grid_anchors(const std::array<double, 3>& center_a,
 }
 
 TimeDependentMapOptions::TimeDependentMapOptions(
-    double initial_time, ExpansionMapOptions expansion_map_options,
-    RotationMapOptions rotation_options,
-    SizeMapOptions<domain::ObjectLabel::A> size_options_A,
-    SizeMapOptions<domain::ObjectLabel::B> size_options_B,
-    ShapeMapOptions<domain::ObjectLabel::A> shape_options_A,
-    ShapeMapOptions<domain::ObjectLabel::B> shape_options_B,
+    double initial_time,
+    std::optional<ExpansionMapOptions> expansion_map_options,
+    std::optional<RotationMapOptions> rotation_options,
+    std::optional<ShapeMapOptions<domain::ObjectLabel::A>> shape_options_A,
+    std::optional<ShapeMapOptions<domain::ObjectLabel::B>> shape_options_B,
     const Options::Context& context)
     : initial_time_(initial_time),
       expansion_map_options_(expansion_map_options),
-      initial_angular_velocity_(rotation_options.initial_angular_velocity),
-      initial_size_values_(std::array{size_options_A.initial_values,
-                                      size_options_B.initial_values}),
-      initial_l_max_{shape_options_A.l_max, shape_options_B.l_max} {
-  const auto check_l_max = [&context](const size_t l_max,
+      rotation_options_(rotation_options),
+      shape_options_A_(shape_options_A),
+      shape_options_B_(shape_options_B) {
+  if (not(expansion_map_options_.has_value() or rotation_options_.has_value() or
+          shape_options_A_.has_value() or shape_options_B_.has_value())) {
+    PARSE_ERROR(context,
+                "Time dependent map options were specified, but all options "
+                "were 'None'. If you don't want time dependent maps, specify "
+                "'None' for the TimeDependentMapOptions. If you want time "
+                "dependent maps, specify options for at least one map.");
```

Comment:
Is there a reason to forbid this?  It seems like it might be easier to experiment if you could toggle all the maps independently.

Reply:
If you set `TimeDependentMapOptions: None`, then time dependent maps are never actually injected into the domain, and the domain is static (`stationary_map` in the block is valida and `block.is_time_dependent()` is false). If you specify all maps as `None` instead (and I remove several errors that are currently in place and add in some identity maps), then you'd have a domain that is in practice time-independent, but the `stationary_map` in the block is nullptr and `block.is_time_dependent()` is true, causing the code to go through all the moving mesh stuff. Even though they'd be identities, I decided this wasn't ideal and to just not allow all individual maps to be `None`.